### ISSUE TITLE
[amp-analytics] missing Support for Session cookies

### ISSUE THREAD

**Issue by Gott50** [ampproject/amphtml#34937]
### Description

Currently it is not possible to create Session cookies, because cookieMaxAge is only implemented to handle Number

Example:
<amp-analytics><script type="application/json">
{
        'cookies': {
          'cookieMaxAge': 604800, // 1 week in seconds
          'aCookie': {
            'value': 'testValue',
          },
        },
}
</script></amp-analytics>

I would like to create a Session cookie. That could look like this:

<amp-analytics><script type="application/json">
{
        'cookies': {
          'cookieMaxAge': 'Session', // Session Cookie
          'aCookie': {
            'value': 'testValue',
          },
        },
}
</script></amp-analytics>

### Reproduction Steps

Try creating a Session cookie. For Example like this:

<amp-analytics><script type="application/json">
{
        'cookies': {
          'cookieMaxAge': 'Session', // Session Cookie
          'aCookie': {
            'value': 'testValue',
          },
        },
}
</script></amp-analytics>

**Comment by micajuine-ho** [comment]
@Gott50 Are you proposing to make the session expiration time indefinite? What is the use case?

**Comment by Gott50** [comment]
> @Gott50 Are you proposing to make the session expiration time indefinite? What is the use case?

@micajuine-ho Yes, I would like to set a cookie without an `expires` value. This is called a session cookie.
https://www.cookiepro.com/knowledge/what-is-a-session-cookie/

Usually I would just set it with JS, like this:
`document.cookie = "cname=cvalue"`
but that is not possible in AMP :/

**Comment by micajuine-ho** [comment]
Is using the default 1year expiration period not desirable?

**Comment by Gott50** [comment]
> Is using the default 1year expiration period not desirable?

unfortunately only Session cookies are allowed without asking the user, because of GDPR
https://en.wikipedia.org/wiki/General_Data_Protection_Regulation

**Comment by micajuine-ho** [comment]
I'm sorry, I'm a little confused. Are you only supposed to set the session cookie after you have consent from the user? Are you using amp-consent?

**Comment by Gott50** [comment]
No, if the user has not consented yet we can only use session cookies, because they are not persistent.
After we have the consent of the user we can use "regular" cookies that last longer than the session (e.g. 1 year)

**Comment by micajuine-ho** [comment]
> No, if the user has not consented, yet, we can only use session cookies, because they are not persistent.

So you want to set a session cookie that is reset every time a user that has not consented enters a page?

What information do you want to store in the cookie?

**Comment by samouri** [comment]
@Gott50: supporting a Session cookie sounds reasonable to me.
@micajuine-ho, do you think this FR should require an I2I or is it small enough to not need one?

**Comment by Gott50** [comment]
> @Gott50: supporting a Session cookie sounds reasonable to me.
> @micajuine-ho, do you think this FR should require an I2I or is it small enough to not need one?

This use case is really necessary in the EU because of GDPR and we need that feature ASAP. That's why I have also implemented the solution already. It is a really simple and short fix.
https://github.com/ampproject/amphtml/pull/34938

Please let me know, if there is anything else I can do to help with the Release of this Feature.

**Comment by micajuine-ho** [comment]
@Gott50  I'm still confused about what this feature entails, looking at your PR, you're setting a cookie with no expiration, and that doesn't really make sense to me, within the context of the CookieWriter module and your use case.

Can you please explain what information do you want to store in the cookie?

@samouri It would at least need to be taken to design review, since we're dealing with storage of data.

**Comment by samouri** [comment]
@micajuine-ho, according to [mdn docs on cookie.set](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/set#parameters), a cookie without an `expirationDate` is a "session cookie".

I agree that this should be brought to design review though, even if just to formalize/agree upon the API.
